conan/tools/build/cpu.py

```diff
@@ -15,8 +16,18 @@ def _cpu_count():
     try:
         try:
             # This is necessary to deduce docker cpu_count
-            cfs_quota_us = int(load("/sys/fs/cgroup/cpu/cpu.cfs_quota_us"))
-            cfs_period_us = int(load("/sys/fs/cgroup/cpu/cpu.cfs_period_us"))
+            # cgroup2
+            if os.path.exists("/sys/fs/cgroup/cgroup.controllers"):
+                cpu_max = load("/sys/fs/cgroup/cpu.max").split()
+                if cpu_max[0] == "max":
+                    return 1
```

Comment:
In case of `max` it means basically no limits are established and therefore we should be able to rely on the `multiprocessing.cpu_count()`. In the current logic you would otherwise effectively in case of docker when cgroup V2 present and without any limiting of CPU cores applied just run with 1 jobs always.

Reply:
But is it not an issue that ``multiprocessing.cpu_count()`` is used? because it will not match the docker assigned cores, and it can also overflow the container capacity?

Reply:
Great, thanks very much for the clarification. @MariaMozgunova could you please apply this in your PR? Thanks!